scalableminds/webknossos
# URL-Sharing of Datasets

### Mailable description of changes:
 - Private Datasets can be shared with anonymous users with a designated token

### Changes Backend
 * new optional parameter for `/api/datasets/:dataSetName`: `sharingToken`
 * new routes `GET` + `DELETE`  `/api/datasets/:dataSetName/sharingToken` (get will create new if none is present)

this sharingToken can also be sent to the datastore (just use the already existing `token`-field there).
Note: the sharingToken will only be used if the user is not logged in.

### TODO: Evolution

### Steps to test:
- Go to edit dataset view
- Generate and copy the sharing link
- logout (maybe turn off auto-login)
- try to open the sharing link (should work)
- try to open with with a different token (shouldn't work) 

### Issues:
- fixes #2435


------
